src/crepe/system/AISystem.cpp

```diff
@@ -0,0 +1,192 @@
+#include <algorithm>
+#include <cmath>
+
+#include "api/LoopTimer.h"
+#include "api/Rigidbody.h"
+#include "api/Transform.h"
+#include "manager/ComponentManager.h"
+#include "manager/Mediator.h"
+
+#include "AISystem.h"
+
+using namespace crepe;
+
+void AISystem::update() {
+	const Mediator & mediator = this->mediator;
+	ComponentManager & mgr = mediator.component_manager;
+	RefVector<AI> ai_components = mgr.get_components_by_type<AI>();
+
+	//TODO: Use fixed loop dt (this is not available at master at the moment)
+	double dt = LoopTimer::get_instance().get_delta_time();
+
+	// Loop through all AI components
+	for (AI & ai : ai_components) {
+		if (!ai.active) {
+			continue;
+		}
+
+		RefVector<Rigidbody> rigidbodies
+			= mgr.get_components_by_id<Rigidbody>(ai.game_object_id);
+		Rigidbody & rigidbody = rigidbodies.front().get();
+
+		// Calculate the force to apply to the entity
+		vec2 force = this->calculate(ai);
+		// Calculate the acceleration (using the above calculated force)
+		vec2 acceleration = force / rigidbody.data.mass;
+		// Finally, update Rigidbody's velocity
+		rigidbody.data.linear_velocity += acceleration * dt;
+	}
+}
+
+vec2 AISystem::calculate(AI & ai) {
```

Comment:
als ik het goed begrijp is nu de prioriteit altijd flee -> arrive -> seek -> path_follow is het misschien een idee dat het mogelijk is voor de user om dit zelf te bepalen?

Reply:
Die prioriteit klopt inderdaad. Het zou inderdaad mooi zijn als de gebruiker dit zelf kan kiezen, maar dat is voor nu niet noodzakelijk volgens mij. Ik zal hier een ticket voor aanmaken in Miro (bij de Nice to have).